server/notification-providers/signl4.js

```diff
@@ -0,0 +1,85 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+const { Liquid } = require("liquidjs");
+const { UP, DOWN } = require("../../src/util");
+
+class SIGNL4 extends NotificationProvider {
+
+    name = "SIGNL4";
+
+    /**
+     * @inheritdoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        let okMsg = "Sent Successfully.";
+
+        try {
+            let data = {
+                heartbeat: heartbeatJSON,
+                monitor: monitorJSON,
+                msg,
+            };
+            let config = {
+                headers: {
+                    "Content-Type": "application/json"
+                }
+            };
+
+            // Source system
+            data["X-S4-SourceSystem"] = "UptimeKuma";
+
+            // Monitor URL
+            let monitorUrl;
+            if (monitorJSON) {
+                if (monitorJSON.type === "port") {
+                    monitorUrl = monitorJSON.hostname;
+                    if (monitorJSON.port) {
+                        monitorUrl += ":" + monitorJSON.port;
+                    }
+                } else if (monitorJSON.hostname != null) {
+                    monitorUrl = monitorJSON.hostname;
+                } else {
+                    monitorUrl = monitorJSON.url;
+                }
+            }
+
+            if (heartbeatJSON == null) {
+                // Test alert
+                data.title = "Uptime Kuma Alert";
+                data.message = "Uptime Kuma Test Alert";
+            } else if (heartbeatJSON.status === UP) {
+                data.title = "Uptime Kuma Monitor ✅ Up";
+                data["X-S4-ExternalID"] = "UptimeKuma" + monitorUrl;
```

Comment:
the monitoring url is likely a bad ID as it could not be set.
I think there should be the id for the monitor in either `heartbeatJSON` or `monitorJSON`

Reply:
@rons4 I think this comment went under your radar.
I'd assume this is a bug, right?

Reply:
Ah, yes. You are right. The ID is much better.  I will change this.